dashboard/data/bmg_plate.py

```diff
@@ -149,6 +149,53 @@ def well_to_ids(well_name: str) -> tuple[int, int]:
     return ord(head) - 65, int(tail) - 1
 
 
+def parse_bmg_file_iostring(filename: str, filecontent: str) -> np.array:
+    """
+    Read data from iostring file to np.array
+
+    :return: array with plate values
+    """
+    plate = np.zeros(shape=(16, 24))
+    barcode = filename.split(".")[0].split("\\")[-1]
+    content = []
+    line = filecontent.readline()
+    while line:
+        content.append(line)
+        line = filecontent.readline()
+    for i, line in enumerate(content):
+        # handle different formatting (as in FIREFLY)
+        if i == 0 and "A" not in line:
+            df = pd.read_csv(content, header=None)
+            plate = df.to_numpy()
+            break
+        well, value = line.split()
+        i, j = well_to_ids(well)
+        plate[i, j] = value
+    return barcode, plate
+
+
+def parse_bmg_files_from_iostring(
```

Comment:
`files` has wrong return type, second `tuple` element is not a `str` + braces `()` are redundant in type definition (should be `tuple[str, *correct_type*]`)